### ISSUE TITLE
Yaml autocompletion does not work in VS Code

### ISSUE THREAD

**Issue by incendial** [Dart-Code/Dart-Code#5320]
**Describe the bug**
Autocompletion for `analysis_options.yaml` (and other yaml files) and not work on typing (but works on a shortcut, `option + esc` in my case).

**To Reproduce**
1. Open any analysis_options file
2. Start typing

**Please complete the following information:**

Dart Code extension: 3.98.1

App: Visual Studio Code
App Host: desktop
Version: mac 1.94.2

Workspace type: Dart (LSP)
Workspace name: dart-code-metrics-teams

Dart (3.5.1): /Users/incendial/Desktop/flutter/bin/cache/dart-sdk
Flutter (3.24.1): /Users/incendial/Desktop/flutter (No device)

**Comment by incendial** [comment]
cc @DanTup 

Related: https://github.com/microsoft/vscode/issues/232363

**Comment by RedCMD** [comment]
did you disable RedHat's YAML extension?

**Comment by incendial** [comment]
> ```json
> "[yaml]": {
> 	"editor.quickSuggestions": {
> 		"strings": "on"
> 	}
> }
> ```

While this solves my particular issue, there needs to be a way for yaml autocompletion to work out of the box as asking millions of dart devs to update their vscode setting is unfeasible.

**Comment by DanTup** [comment]
Yes, I don't think changing the settings is a good fix (either for users to do, or for the Dart extension to override). Unless there is a deliberate reason for it to work like this in VS Code (and I'm not thinking of one... many users think code completion is completely broken if it doesn't pop up automatically where they would expect), I feel like something needs to change there.

Personally I feel like either these things are not "strings", or the setting above should be default for YAML in VS Code...  for other languages, strings means things the user would never expected to get completion on (they're just literal strings) though and I don't think that really applies for YAML (because pretty much anything, key or value, could get reasonable completion).

**Comment by RedCMD** [comment]
> Yes, I don't think changing the settings is a good fix

thats a workaround for https://github.com/microsoft/vscode/issues/231224
user settings don't seem to be affected

**Comment by DanTup** [comment]
> as you can see the setting is already enabled for YAML (via RedHat)

Users might not have the RedHat extension.

I don't think it makes sense to require any extension that wants to provide any YAML suggestions to also have to override this setting.. Either it should not be required, or VS Code should have a sensible default.

In this case, we also only supply completion for a small number of YAML files, so changing the users settings for all YAML files isn't appropriate anyway.

**Comment by DanTup** [comment]
Since the previous issue was closed, I've opened https://github.com/microsoft/vscode/issues/239679 about about trying to change this in VS Code. I still don't feel it makes sense for Dart to change settings for all YAML files, and I still feel that VS Code's default here is bad.

**Comment by RedCMD** [comment]
I can't find any yaml completion code in this repo
https://github.com/Dart-Code/Dart-Code/blob/master/src/extension/providers/snippet_completion_item_provider.ts
is it somewhere else?

there is a temp [fix](https://github.com/microsoft/vscode/pull/240422) in [insiders](https://insiders.vscode.dev/) for https://github.com/microsoft/vscode/issues/231224

**Comment by DanTup** [comment]
@RedCMD the completion comes from the LSP server, which is part of the Dart SDK.

> there is a temp https://github.com/microsoft/vscode/pull/240422 in [insiders](https://insiders.vscode.dev/) for https://github.com/microsoft/vscode/issues/231224

This doesn't solve the problem, we don't set any `configurationDefaults` because a) I don't think it's right for Dart to change settings for all YAML files just because it provides completion for two specific YAML files (`pubspec.yaml` and `analysis_options.yaml`), and I also think the VS Code default is bad. It doesn't make sense to say that any extension providing YAML completion should override this setting - the default should just be fixed to support the behaviour that users want/expect.

**Comment by DanTup** [comment]
This is still blocked on https://github.com/microsoft/vscode/issues/239679 - I still believe this is an issue that should be fixed in VS Code. Please add 👍 's that issue.

**Comment by DanTup** [comment]
This has been fixed in VS Code for the next release - thanks @RedCMD!
